### PR TITLE
Th java language server

### PR DESCRIPTION
**Pull request by airportyh**
Why
===

Switch to https://github.com/georgewfraser/java-language-server for less memory and cpu consumption. Get this
[configuration feature in](https://github.com/replit/java-language-server/pull/1).

What changed
============

1. Added nix file for java-language-server
_Describe what changed to a level of detail that someone with no context with your PR could be able to review it_

Test plan
=========

Await https://github.com/replit/java-language-server/pull/1 to be merged so we can updated the revs and shas
1. Make blank repl
2. install the Java Graalvm module
3. Make a swing program `Main.java` based on https://gist.github.com/airportyh/b97570b1c46d839678ebd14e6e8e690d
4. See that you can hover over `JFrame` and get info on it

### CODE REVIEW

File: pkgs/java-language-server/default.nix
[the remark this answers is not in the document]
Comment (airportyh) [reply pkgs/java-language-server/default.nix:1]: True, but I could see benefit of keeping this here too. It is modified from https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/development/tools/java/java-language-server/default.nix so we can diff it against upstream if needed.

@ryantm do you have a preference here?

File: flake.nix [hunk 1]
```diff
@@ -3,12 +3,13 @@
   inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-23.05";
   inputs.nixpkgs-unstable.url = "github:nixos/nixpkgs/nixpkgs-unstable";
   inputs.prybar.url = "github:replit/prybar";
+  inputs.java-language-server.url = "github:replit/java-language-server";
```
Comment (ryantm) [inline flake.nix:6]:
```suggestion
  inputs.java-language-server.url = "github:replit/java-language-server";
  inputs.java-language-server.inputs.nixpkgs.follows = "nixpkgs";
```
Ideally, we use the same nixpkgs as everything else  or "nixpkgs-unstable" to limit the number of nixpkgs checkouts we have to evaluate.

File: pkgs/modules/java/default.nix [hunk 2]
```diff
@@ -1,19 +1,31 @@
-{ pkgs, lib, ... }:
+{ pkgs-unstable, lib, ... }:
 
 let
-  graalvm = pkgs.graalvm17-ce;
+  pkgs = pkgs-unstable;
+  graalvm = pkgs.graalvm19-ce;
```
Comment (ryantm) [inline pkgs/modules/java/default.nix:5]: So weird that  "19-ce" in "graalvm19-ce" is version "22.3", but that seems right:

image

File: pkgs/modules/java/default.nix [hunk 3]
```diff
@@ -1,19 +1,31 @@
-{ pkgs, lib, ... }:
+{ pkgs-unstable, lib, ... }:
```
Comment (ryantm) [inline pkgs/modules/java/default.nix:1]: Why do we need to use unstable?

**Review by ryantm — APPROVED** [review]

File: pkgs/modules/java/default.nix (hunk 3 above)
Comment (airportyh) [reply pkgs/modules/java/default.nix:1]: The stable one didn't have version 19 of graalvm, which is needed to compile the project.

Comment (airportyh) [reply pkgs/modules/java/default.nix:1]: Now that we switched to 23.05 probably don't need to.

**Pull request merged**